Crashes at wakeup

Bunch of reports with this stacktrace. Might be happening because we don't persist password, and during wakeup its gone from memory.

```
java.lang.RuntimeException: Unable to start activity ComponentInfo{com.twistedplane.sealnote/com.twistedplane.sealnote.NoteActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String[] java.lang.String.split(java.lang.String)' on a null object reference
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3319)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3415)
	at android.app.ActivityThread.access$1100(ActivityThread.java:229)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1821)
	at android.os.Handler.dispatchMessage(Handler.java:102)
	at android.os.Looper.loop(Looper.java:148)
	at android.app.ActivityThread.main(ActivityThread.java:7325)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1230)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1120)
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String[] java.lang.String.split(java.lang.String)' on a null object reference
	at com.twistedplane.sealnote.data.Note.convertToTagSet(Note.java:118)
	at com.twistedplane.sealnote.data.Note.readFromParcel(Note.java:97)
	at com.twistedplane.sealnote.data.Note.<init>(Note.java:76)
	at com.twistedplane.sealnote.data.Note$1.createFromParcel(Note.java:250)
	at com.twistedplane.sealnote.data.Note$1.createFromParcel(Note.java:248)
	at android.os.Parcel.readParcelable(Parcel.java:2367)
	at android.os.Parcel.readValue(Parcel.java:2264)
	at android.os.Parcel.readArrayMapInternal(Parcel.java:2614)
	at android.os.BaseBundle.unparcel(BaseBundle.java:221)
	at android.os.BaseBundle.getBoolean(BaseBundle.java:659)
	at android.app.Activity.restoreHasCurrentPermissionRequest(Activity.java:7153)
	at android.app.Activity.performCreate(Activity.java:6903)
	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1136)
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3266)
	... 9 more
```


```
java.lang.RuntimeException: An error occured while executing doInBackground()
	at android.os.AsyncTask$3.done(AsyncTask.java:304)
	at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:355)
	at java.util.concurrent.FutureTask.setException(FutureTask.java:222)
	at java.util.concurrent.FutureTask.run(FutureTask.java:242)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
	at java.lang.Thread.run(Thread.java:818)
Caused by: java.lang.IllegalArgumentException: Password null or not acceptable
	at com.twistedplane.sealnote.data.DatabaseHandler.update(DatabaseHandler.java:84)
	at com.twistedplane.sealnote.data.DatabaseHandler.getWritableDatabase(DatabaseHandler.java:110)
	at com.twistedplane.sealnote.data.DatabaseHandler.getReadableDatabase(DatabaseHandler.java:115)
	at com.twistedplane.sealnote.data.DatabaseHandler.getNotesCursor(DatabaseHandler.java:558)
	at com.twistedplane.sealnote.data.AdapterLoader.loadInBackground(AdapterLoader.java:30)
	at com.twistedplane.sealnote.data.AdapterLoader.loadInBackground(AdapterLoader.java:13)
	at android.content.AsyncTaskLoader.onLoadInBackground(AsyncTaskLoader.java:312)
	at android.content.AsyncTaskLoader$LoadTask.doInBackground(AsyncTaskLoader.java:69)
	at android.content.AsyncTaskLoader$LoadTask.doInBackground(AsyncTaskLoader.java:57)
	at android.os.AsyncTask$2.call(AsyncTask.java:292)
	at java.util.concurrent.FutureTask.run(FutureTask.java:237)
	... 3 more
```

--- a/Sealnote/src/main/java/com/twistedplane/sealnote/NoteActivity.java
+++ b/Sealnote/src/main/java/com/twistedplane/sealnote/NoteActivity.java
@@ -107,6 +107,12 @@ public class NoteActivity extends Activity implements ColorDialogFragment.ColorC
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_note);
+
+        if (!Misc.isPasswordLoaded()){
+            Misc.startPasswordActivity(this);
+            return;
+        }
+
         // Even though we change content view later, we secure window as soon as possible
         Misc.secureWindow(NoteActivity.this);
 

--- a/Sealnote/src/main/java/com/twistedplane/sealnote/data/Note.java
+++ b/Sealnote/src/main/java/com/twistedplane/sealnote/data/Note.java
@@ -83,7 +83,8 @@ public class Note implements Parcelable{
         mId = inParcel.readInt();
         mPosition = inParcel.readInt();
         mNoteTitle = inParcel.readString();
-        String note = inParcel.readString();
+        mType = Type.valueOf(inParcel.readString());
+        mNote = NoteContent.fromString(mType, inParcel.readString());
         try {
             mEditedDate = EasyDate.fromIsoString(inParcel.readString());
         } catch (ParseException e) {
@@ -92,8 +93,6 @@ public class Note implements Parcelable{
         mColor = inParcel.readInt();
         mArchived = inParcel.readInt() > 0;
         mDeleted = inParcel.readInt() > 0;
-        mType = Type.valueOf(inParcel.readString());
-        mNote = NoteContent.fromString(mType, inParcel.readString());
         mTags = convertToTagSet(inParcel.readString());
     }
 
@@ -102,12 +101,12 @@ public class Note implements Parcelable{
         outParcel.writeInt(mId);
         outParcel.writeInt(mPosition);
         outParcel.writeString(mNoteTitle);
+        outParcel.writeString(mType.name());
         outParcel.writeString(mNote.toString());
         outParcel.writeString(mEditedDate.toString());
         outParcel.writeInt(mColor);
         outParcel.writeInt(mArchived ?1 :0);
         outParcel.writeInt(mDeleted ?1 :0);
-        outParcel.writeString(mType.name());
         outParcel.writeString(convertTagSetToString(mTags));
     }
 

--- a/Sealnote/src/main/java/com/twistedplane/sealnote/fragment/SealnoteFragment.java
+++ b/Sealnote/src/main/java/com/twistedplane/sealnote/fragment/SealnoteFragment.java
@@ -18,6 +18,7 @@ import com.twistedplane.sealnote.R;
 import com.twistedplane.sealnote.data.AdapterLoader;
 import com.twistedplane.sealnote.data.Note;
 import com.twistedplane.sealnote.data.SealnoteAdapter;
+import com.twistedplane.sealnote.utils.Misc;
 
 /**
  * Main fragment where all cards are listed in a staggered grid
@@ -78,6 +79,11 @@ abstract public class SealnoteFragment extends Fragment implements
     public void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         Log.d(TAG, "Creating SealNote fragment...");
+
+        if (!Misc.isPasswordLoaded()){
+            Misc.startPasswordActivity(getActivity());
+            return;
+        }
 
         // Get folder active in activity
         String folder = getArguments().getString("SN_FOLDER", Note.Folder.FOLDER_LIVE.name());

--- a/Sealnote/src/main/java/com/twistedplane/sealnote/utils/Misc.java
+++ b/Sealnote/src/main/java/com/twistedplane/sealnote/utils/Misc.java
@@ -2,10 +2,13 @@ package com.twistedplane.sealnote.utils;
 
 import android.app.Activity;
 import android.content.Context;
+import android.content.Intent;
 import android.content.res.Resources;
 import android.view.Window;
 import android.view.WindowManager;
+import com.twistedplane.sealnote.PasswordActivity;
 import com.twistedplane.sealnote.R;
+import com.twistedplane.sealnote.SealnoteApplication;
 
 /**
  * Miscellaneous helper functions
@@ -65,5 +68,15 @@ public class Misc {
         } else {
             window.setFlags(0, WindowManager.LayoutParams.FLAG_SECURE);
         }
+    }
+
+    public static boolean isPasswordLoaded() {
+        return SealnoteApplication.getDatabase().getPassword() != null;
+    }
+
+    public static void startPasswordActivity(Activity activity) {
+        Intent passwordIntent = new Intent(activity, PasswordActivity.class);
+        activity.startActivity(passwordIntent);
+        activity.finish();
     }
 }

